models/dns.tsp

```diff
@@ -0,0 +1,81 @@
+
+namespace rpp.dns {
+
+    @doc("DNS configuration of an object")
+    model DNS {
+      dns?: {
+        records: DnsRecord[];
+        controls?: DnsControls;
+      }
+    }
+
+    @doc("DNS resource record, as specified in RFC 1035. RDATA is added depending on type")
+    model DnsRecord {
+      name: string;
+      type: string;
+      rdata: RDataNS | RDataA | RDataAAAA | RDataDS | RDataDNSKEY | RDataMX | RDataTXT
+    }
+
+    @doc("RDATA for NS record, as specified in RFC 1035")
+    model RDataNS extends DnsRecord {
```

Comment:
"extends DnsRecord" does not seem correct to me.
It generates "name/type/rdata" fields, where rdata is again RData* inherited from DnsRecord... so it goes indefinitely.

Reply:
right, also applies to the other RR Types.